### PR TITLE
[KV Cache Injection] Causal Mask for CodeGen

### PR DESCRIPTION
**Pull request by dbogunowicz**
This PR:

- implements causal mask injection for the CodeGen model
- begins a process of refactoring the `AdditionalTransforms` objects, so that the `AdditionalTransformsBase` is to share more and more helper methods that can be shared across its children classes and reduce boilerplate code. This process will be continued with the incoming causal mask injections for other LLMs

For testing please refer to: https://github.com/neuralmagic/deepsparse/pull/1127

### CODE REVIEW

File: src/sparseml/pytorch/utils/helpers.py [hunk 1]
```diff
@@ -1174,3 +1177,41 @@ def detach(x: Union[torch.Tensor, List, Tuple]):
         return tuple([detach(e) for e in x])
     else:
         raise ValueError("Unexpected type to detach")
+
+
+def adjust_quantization_for_onnx_export(module: torch.nn.Module) -> torch.nn.Module:
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/helpers.py:1182]: this needs rebase
